### PR TITLE
Bws47

### PR DESCRIPTION
**Pull request by cmgustavo**
* Improves list of transactions history
* Cache for history
* Mobile look&feel for modals of transactions history.

### CODE REVIEW

File: src/js/controllers/history.js [hunk 1]
```diff
@@ -26,9 +28,30 @@ angular.module('copayApp.controllers').controller('historyController',
       return this.alternativeIsoCode;
     };
 
-    this.getTxHistory = function() {
+    this.checkCacheTxHistory = function() {
```
Comment (matiu) [inline src/js/controllers/history.js]: shouldn't be 'check' since it sets self.txHistory from the cache.

File: src/js/controllers/history.js [hunk 2]
```diff
@@ -36,21 +59,39 @@ angular.module('copayApp.controllers').controller('historyController',
...
+                c++;
+              }
             });
 
-            self.txHistory = txs;
+            if (firstTime) {
+              self.storeCacheTxHistory(JSON.stringify(self.txHistory));
+            }
+
+            self.skip = self.skip + self.limit;
```
Comment (matiu) [inline src/js/controllers/history.js]: self.skip += self.limit

File: src/js/controllers/history.js [hunk 3]
```diff
@@ -36,21 +59,39 @@ angular.module('copayApp.controllers').controller('historyController',
         }, function(err, txs) {
           if (err) {
             $log.debug('Creating address ERROR:', err);
+            $scope.$emit('Local/ClientError', err);
           }
           else {
+
+            if (firstTime) {
+              self.txHistory = [];
```
Comment (matiu) [inline src/js/controllers/history.js]: Can we just put self.txHistory = [] at the begining of the controller?

File: src/js/controllers/history.js [hunk 4]
```diff
@@ -26,9 +28,30 @@ angular.module('copayApp.controllers').controller('historyController',
...
+    this.storeCacheTxHistory = function(txps) { 
+      storageService.storeLastTransactions(fc.credentials.walletId, txps, function(err) {
+        if (err) $log.debug('Error: ', err);
+      });
+    };
+
+    this.getTxHistory = function(firstTime) {
+      var self = this;
+      if (firstTime) {
+        self.checkCacheTxHistory();
```
Comment (matiu) [inline src/js/controllers/history.js]: checkCAchekTxHistory is async, it should have a callback.

File: src/js/controllers/history.js [hunk 5]
```diff
@@ -26,9 +28,30 @@ angular.module('copayApp.controllers').controller('historyController',
...
+      }
       self.updatingTxHistory = true;
+      self.loadMore = false;
       $timeout(function() {
         fc.getTxHistory({
```
Comment (matiu) [inline src/js/controllers/history.js]: I dont get why getTxHistory is called always, even if self.txHistory is set from the cache.

**Comment by matiu** [comment]
Also, cache should have a 'clearCache' function, called with new incoming /outgoing tx are made.

**Comment by cmgustavo** [comment]
Thanks for your comment! Finally we agreed to save the first page of history on memory, not in localStorage and trigger update if receives or sends funds.

**Comment by cmgustavo** [comment]
Updated

**Pull request merged**